forte/data/entry_converter.py

```diff
@@ -0,0 +1,209 @@
+        data_store_ref = pack._data_store  # pylint: disable=protected-access
+        try:
+            data_store_ref.get_entry(tid=entry.tid)
+            logger.info(
+                "The entry with tid=%d is already saved into DataStore",
+                entry.tid,
+            )
+            return
+        except KeyError:
+            # The entry is not found in DataStore
+            pass
+
+        # Create a new registry in DataStore based on entry's type
+        if isinstance(entry, Annotation):
+            data_store_ref.add_annotation_raw(
+                type_name=entry.entry_type(),
+                begin=entry.begin,
+                end=entry.end,
+                tid=entry.tid,
+                allow_duplicate=allow_duplicate,
+            )
+        elif isinstance(entry, Link):
+            data_store_ref.add_link_raw(
+                type_name=entry.entry_type(),
+                parent_tid=entry.parent,
+                child_tid=entry.child,
+                tid=entry.tid,
+            )
+        elif isinstance(entry, Group):
+            data_store_ref.add_group_raw(
+                type_name=entry.entry_type(),
+                member_type=get_full_module_name(entry.MemberType),
+                tid=entry.tid,
+            )
+        elif isinstance(entry, Generics):
+            data_store_ref.add_generics_raw(
+                type_name=entry.entry_type(),
+                tid=entry.tid,
+            )
+        elif isinstance(entry, AudioAnnotation):
+            data_store_ref.add_audio_annotation_raw(
+                type_name=entry.entry_type(),
+                begin=entry.begin,
+                end=entry.end,
+                tid=entry.tid,
+                allow_duplicate=allow_duplicate,
+            )
+        elif isinstance(entry, ImageAnnotation):
+            data_store_ref.add_image_annotation_raw(
+                type_name=entry.entry_type(),
+                image_payload_idx=entry.image_payload_idx,
+                tid=entry.tid,
+            )
+        elif isinstance(entry, Grids):
+            data_store_ref.add_grid_raw(
+                type_name=entry.entry_type(),
+                image_payload_idx=entry.image_payload_idx,
+                tid=entry.tid,
+            )
+        elif isinstance(entry, MultiPackLink):
+            data_store_ref.add_multipack_link_raw(
+                type_name=entry.entry_type(),
+                parent_pack_id=entry.parent[0],
+                parent_tid=entry.parent[1],
+                child_pack_id=entry.child[0],
+                child_tid=entry.child[1],
+                tid=entry.tid,
+            )
+        elif isinstance(entry, MultiPackGroup):
+            data_store_ref.add_multipack_group_raw(
+                type_name=entry.entry_type(),
+                member_type=get_full_module_name(entry.MemberType),
+                tid=entry.tid,
+            )
+        elif isinstance(entry, MultiPackGeneric):
```

Comment:
Now here we will have more places to update whenever we create a new top-level type. We should do somethign about it, we either can:
1. document what needs to be updated when adding a new top level type in `top.py`
2. find a way to centralize this

And we should also make sure our test cases cover these when adding a new type.